numpy/lib/utils.py

```diff
@@ -904,7 +904,7 @@ def _lookfor_generate_cache(module, import_modules, regenerate):
                                 sys.stdout = old_stdout
                                 sys.stderr = old_stderr
                         # Catch SystemExit, too
-                        except BaseException:
+                        except SystemExit:
```

Comment:
This change will now miss any other type of error like ImportError or IOError.

Reply:
```suggestion
                        except (Exception, SystemExit):
```

Reply:
Thanks, I had missed that